### ISSUE TITLE
undefined reference to `dt_getprop_as_number' during linking process

### ISSUE THREAD

**Issue by mardyr** [OP-TEE/optee_os#6537]
Hi, guys

I am on latest commit 0b9aa2785

I've cloned the repo, had built the toolchain based on instructions and built os with instruction
```makefile
make \
    CFG_ARM64_core=y \
    CFG_TEE_BENCHMARK=n \
    CFG_TEE_CORE_LOG_LEVEL=0 \
    CROSS_COMPILE=aarch64-linux-gnu- \
    CROSS_COMPILE_core=aarch64-linux-gnu- \
    CROSS_COMPILE_ta_arm32=arm-linux-gnueabihf- \
    CROSS_COMPILE_ta_arm64=aarch64-linux-gnu- \
    DEBUG=1 \
    O=out/arm \
    PLATFORM=vexpress-juno
``` 
During linking process I've got this errors ```arch64-linux-gnu-ld.bfd: out/arm/core/arch/arm/kernel/boot.o: in function `get_sec_mem_from_manifest':
/home/mqj684/optee-repos/optee_os/core/arch/arm/kernel/boot.c:1500: undefined reference to `dt_getprop_as_number'
aarch64-linux-gnu-ld.bfd: /home/mqj684/optee-repos/optee_os/core/arch/arm/kernel/boot.c:1508: undefined reference to `dt_getprop_as_number'
aarch64-linux-gnu-ld.bfd: out/arm/core/mm/core_mmu.o: in function `collect_device_mem_ranges':
/home/mqj684/optee-repos/optee_os/core/mm/core_mmu.c:980: undefined reference to `dt_getprop_as_number'
aarch64-linux-gnu-ld.bfd: /home/mqj684/optee-repos/optee_os/core/mm/core_mmu.c:991: undefined reference to `dt_getprop_as_number'
make: *** [core/arch/arm/kernel/link.mk:69: out/arm/core/all_objs.o] Error 1``` 

which means that there are problems with finding symbols, well I've looked into both of source files and symbols are included `#include <kernel/dt.h>` so it has to be some makefile nightmare that paths are lost somewhere, of course I have this files on repo ```optee_os]$ find -name "dt.h"
./core/include/kernel/dt.h```

**Comment by mardyr** [comment]
funny thing, I've remove DEBUG flag and it compiles, so under this flag some includes are not passed to sources , it could be addressed :)

**Comment by Christine7414** [comment]
It works for me by adding the `$(call force,CFG_DT,y)` in the platform specific `conf.mk`.

**Comment by mlhh1103** [comment]
hi, I've removed DEBUG flag and the optee_os is compiled, but I got this error when I "make" the optee_client.
root@mlh-virtual-machine:/home/mlh/optee/optee_client# make CROSS_COMPILE=aarch64-linux-gnu-
Building libteec.so
Building tee-supplicant
make --directory=tee-supplicant  --no-print-directory --no-builtin-variables CFG_TEE_SUPP_LOG_LEVEL=1
make[1]: Nothing to be done for 'all'.
Building libckteec.so
Building libseteec.so
Building libteeacl.so
make[1]: aarch64-linux-gnu-pkg-config: No such file or directory
make[1]: aarch64-linux-gnu-pkg-config: No such file or directory
  Finding uuid.pc
make[1]: aarch64-linux-gnu-pkg-config: No such file or directory
make[1]: *** [Makefile:47: check-libuuid] Error 127
make: *** [Makefile:58: build-libteeacl] Error 2

**Comment by mardyr** [comment]
That's different repo, so it shoudln't be addressed here, but here is a workaround https://github.com/OP-TEE/optee_client/issues/368

**Comment by nikiyadav** [comment]
Hi,

I am facing the same issue while building OPTEE with FVP. I ran the make command with DEBUG=1 and got linking errors. I set CFG_DT=y flag in the make command too and after that I get the following output:

<img width="1512" alt="Screenshot 2023-12-28 at 3 42 19 AM" src="https://github.com/OP-TEE/optee_os/assets/10370377/9f403fca-47c0-4072-9115-6c42e0a0ab36">

The boot process gets stuck at initialising BL32. Any idea what is wrong and how do I resolve this? Thanks.

**Comment by nikiyadav** [comment]
Hi,

I am facing the same issue while building OPTEE with FVP. I ran the make command with DEBUG=1 and got linking errors. I set CFG_DT=y flag in the make command too and after that I get the following output:

<img width="795" alt="Screenshot 2023-12-28 at 3 45 08 AM" src="https://github.com/OP-TEE/optee_os/assets/10370377/49841e74-7795-437b-800c-64321095eaa5">

The boot process is stuck at initialising BL32. Any idea what is wrong and how do I resolve it? Is it wrong to set CFG_DT=y in the make command? Thanks.
